pkg/notary/notary_test.go

```diff
@@ -4,18 +4,18 @@ import (
 	"context"
 	"testing"
 
+	gcrremote "github.com/google/go-containerregistry/0_14/pkg/v1/remote" // TODO: Remove this once we upgrade tp cosign version < 2.0.2
```

Comment:
what's this ?

Reply:
While upgrading to `Cosign 2.1.1`,  I faced this issue with k8s-manifest-sigstore
```
github.com/sigstore/k8s-manifest-sigstore@v0.5.0/pkg/k8smanifest/provenance.go:569:32: undefined: rekorutil.PublicKey
```
I believe that is because `Cosign 2.0.2` uses `Rekor 1.1` and `Cosign 2.1.1 `uses `Rekor 1.2`

Current notary implementation uses gcr remote's Reuse function and cosign 2.0.2 doesn't allow us to upgrade to the latest version of gcr remote. So I am maintaining 2 versions of gcr remote, unit we can to `Cosign 2.1.1` to support it.

I have asked the maintainers of k8s-manifest-sigstore about it https://sigstore.slack.com/archives/C01DGF0G8U9/p1688724765487689

Reply:
Ok, we need to track this